pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -2,156 +2,327 @@ package configurableinterpreter
+		return
 	}
 
-	luaScript := customAccessor.GetStatusAggregationLuaScript()
-	klog.Infof("lua script %s", luaScript)
+	rets, err := eng.Invoke(object)
+	if err != nil {
+		return
+	}
 
-	return nil, err
+	if r := rets[0]; !r.IsNil() {
+		status = &runtime.RawExtension{}
+		err = r.Into(status)
+	}
+	return
 }
 
 // InterpretHealth returns the health state of the object.
-func (c *ConfigurableInterpreter) InterpretHealth(object *unstructured.Unstructured) (bool, error) {
-	klog.Infof("ConfigurableInterpreter Execute InterpretHealth")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetHealthInterpretationLuaScript()) == 0 {
-		return false, fmt.Errorf("customized interpreter GetHealthInterpretation for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) InterpretHealth(object *unstructured.Unstructured) (healthy bool, enabled bool, err error) {
+	eng, enabled := c.getInterpreter(object.GroupVersionKind(), configv1alpha1.InterpreterOperationInterpretHealth)
+	if !enabled {
+		return
 	}
 
-	luaScript := customAccessor.GetHealthInterpretationLuaScript()
-	klog.Infof("lua script %s", luaScript)
+	rets, err := eng.Invoke(object)
+	if err != nil {
+		return
+	}
 
-	return false, nil
+	healthy, err = rets[0].Bool()
+	return
+}
+
+// getInterpreter returns interpreter for specific resource kind and operation.
+func (c *ConfigurableInterpreter) getInterpreter(kind schema.GroupVersionKind, operation configv1alpha1.InterpreterOperation) (engine.Function, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	interpreter, ok := c.interpreters[kind][operation]
+	return interpreter, ok
+}
+
+func (c *ConfigurableInterpreter) updateConfiguration() {
+	objs, err := c.configLister.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("error updating configuration: %v", err))
+		return
+	}
+	configs := make([]*configv1alpha1.ResourceInterpreterCustomization, len(objs))
+	for i, obj := range objs {
+		config := &configv1alpha1.ResourceInterpreterCustomization{}
+		if err = helper.ConvertToTypedObject(obj, config); err != nil {
+			klog.Errorf("Failed to transform ResourceInterpreterCustomization: %v", err)
+			return
+		}
+		configs[i] = config
+	}
+
+	err = c.LoadConfig(configs)
+	if err != nil {
+		klog.Error(err)
+	}
+}
+
+// LoadConfig load customization rules.
+func (c *ConfigurableInterpreter) LoadConfig(configs []*configv1alpha1.ResourceInterpreterCustomization) error {
+	interpreters := map[schema.GroupVersionKind]map[configv1alpha1.InterpreterOperation]engine.Function{}
+	for _, customization := range configs {
+		gv, err := schema.ParseGroupVersion(customization.Spec.Target.APIVersion)
+		if err != nil {
+			return err
+		}
+		kind := gv.WithKind(customization.Spec.Target.Kind)
+		interps, ok := interpreters[kind]
+		if !ok {
+			interps = map[configv1alpha1.InterpreterOperation]engine.Function{}
+			interpreters[kind] = interps
+		}
```

Comment:
This logic assumes all rules are contained in a `single` `ResourceInterpreterCustomization` object. 
What if the rules are declared by several objects?
Such as ResourceInterpreterCustomization-`A` declares the rule of `Retention` and ResourceInterpreterCustomization-`B` declares the rule of `StatusReflection`.
With this logic, only the first observed object can be cached, and the other would be accidentally ignored without any warning message.

Reply:
They are merged. This case is tested in:

https://github.com/karmada-io/karmada/blob/1c840eac2e52c36df20fbfc0893fd753c10c5491/pkg/resourceinterpreter/configurableinterpreter/configurable_test.go#L104-L111

Reply:
get it.